lib/cocoapods/sources_manager.rb

```diff
@@ -206,18 +206,16 @@ def update(source_name = nil, show_output = false)
           sources =  git_sources
         end
 
-        sources.each do |source|
+        WorkerPool.process(sources, 4) do |source|
           UI.section "Updating spec repo `#{source.name}`" do
-            Dir.chdir(source.repo) do
-              begin
-                output = git!('pull --ff-only')
-                UI.puts output if show_output && !config.verbose?
-              rescue Informative
-                UI.warn 'CocoaPods was not able to update the ' \
-                  "`#{source.name}` repo. If this is an unexpected issue " \
-                  'and persists you can inspect it running ' \
-                  '`pod repo update --verbose`'
-              end
+            begin
+              output = git!("--git-dir=#{(source.repo + '.git').to_s.shellescape} --work-tree=#{source.repo.to_s.shellescape} pull --ff-only")
```

Comment:
Maybe it would make sense to unify out `git!` usage and make `git!` deal with `--git-dir`.

```ruby
git!('pull --ff-only', source.repo)
```

Then all the other git usages will not need to chdir either.